### PR TITLE
cucumber tests, but the state of objects is in yaml files

### PR DESCRIPTION
**Pull request by n3wscott**
Another progress PR for integration tests based on table tests built by cucumber.

### CODE REVIEW

File: pkg/reconciler/clusterducktype/testdata/config/smoke/initial.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    name: "Zhangchas"
+    plural: "zhangchas"
+    singular: "zhangcha"
+
+  versions:
+    - name: "v1"
+  group: duck.knative.dev
+
+status:
+  observedGeneration: 0
```
Comment (mattmoor) [inline pkg/reconciler/clusterducktype/testdata/config/smoke/initial.yaml:20]: Add trailing newline:
```suggestion
  observedGeneration: 0

```

File: pkg/reconciler/clusterducktype/testdata/config/smoke/updated.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  versions:
+    - name: "v1"
+  group: duck.knative.dev
+
+status:
+  observedGeneration: {{ .generation }}
+  conditions:
+    - type: Ready
+      status: "True"
```
Comment (mattmoor) [inline pkg/reconciler/clusterducktype/testdata/config/smoke/updated.yaml:23]: Add trailing newline:
```suggestion
      status: "True"

```

File: pkg/reconciler/clusterducktype/clusterducktype_test.go [hunk 3]
```diff
@@ -17,10 +17,15 @@ limitations under the License.
 package clusterducktype
 
 import (
+	"bufio"
 	"context"
 	"flag"
 	"fmt"
+	"github.com/n3wscott/rigging/pkg/installer"
```
Comment (mattmoor) [inline pkg/reconciler/clusterducktype/clusterducktype_test.go:24]: Format Go code:
```suggestion

```

File: pkg/reconciler/clusterducktype/clusterducktype_test.go [hunk 4]
```diff
@@ -17,10 +17,15 @@ limitations under the License.
...
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
```
Comment (mattmoor) [inline pkg/reconciler/clusterducktype/clusterducktype_test.go:26]: Format Go code:
```suggestion

```

File: pkg/reconciler/clusterducktype/clusterducktype_test.go [hunk 5]
```diff
@@ -17,10 +17,15 @@ limitations under the License.
...
 	"os"
+	"path"
 	"strings"
 	"testing"
```
Comment (mattmoor) [inline pkg/reconciler/clusterducktype/clusterducktype_test.go:30]: Format Go code:
```suggestion
	"testing"

	"github.com/n3wscott/rigging/pkg/installer"
	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
```

File: pkg/reconciler/clusterducktype/testdata/config/smoke/initial.yaml (hunk 1 above)
Comment (mattmoor) [inline pkg/reconciler/clusterducktype/testdata/config/smoke/initial.yaml:20]: Add trailing newline:
```suggestion
  observedGeneration: 0

```

File: pkg/reconciler/clusterducktype/testdata/config/smoke/updated.yaml (hunk 2 above)
Comment (mattmoor) [inline pkg/reconciler/clusterducktype/testdata/config/smoke/updated.yaml:23]: Add trailing newline:
```suggestion
      status: "True"

```

File: pkg/reconciler/clusterducktype/testdata/config/smoke/initial.yaml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    - labelSelector: "duck.knative.dev/smoked=true"
+
+  names:
+    name: "Zhangchas"
+    plural: "zhangchas"
+    singular: "zhangcha"
+
+  versions:
+    - name: "v1"
+  group: duck.knative.dev
```
Comment (antoineco) [inline pkg/reconciler/clusterducktype/testdata/config/smoke/initial.yaml:17]: yuck, some many quotes in a YAML 😛

File: pkg/reconciler/clusterducktype/testdata/features/smoke.feature [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+            | file                      | generation   |
+            | config/smoke/initial.yaml | <generation> |
+
+        And a ClusterDuckType reconciler
+
+        When reconciling "zhangchas.duck.knative.dev"
+
+        Then expect status updates (from file):
+            | file                      | generation   |
+            | config/smoke/updated.yaml | <generation> |
```
Comment (antoineco) [inline pkg/reconciler/clusterducktype/testdata/features/smoke.feature:15]: Neat!

File: pkg/reconciler/clusterducktype/clusterducktype_test.go [hunk 8]
```diff
@@ -17,13 +17,19 @@ limitations under the License.
...
 	"fmt"
+	"io/ioutil"
 	"os"
+	"path"
 	"strings"
 	"testing"
 
+	"github.com/n3wscott/rigging/pkg/installer"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
```
Comment (antoineco) [inline pkg/reconciler/clusterducktype/clusterducktype_test.go:32]: Those deserve to be grouped with the other third-party imports.

**Review by antoineco — COMMENTED** [review]
I like it 👍  I'll give someone else a chance to take a look.

File: pkg/reconciler/clusterducktype/testdata/features/smoke.feature (hunk 7 above)
Comment (lionelvillard) [reply pkg/reconciler/clusterducktype/testdata/features/smoke.feature:15]: How flexible is this? Can you say "Given this .. Then that ... Then Delete this .. then that ..." etc.. ?

Comment (n3wscott) [reply pkg/reconciler/clusterducktype/testdata/features/smoke.feature:15]: all things after file in the headers define how the `{{ .name }}` will work, so here, generation is not defined in my test code, but is in the yaml.

File: pkg/reconciler/clusterducktype/clusterducktype_test.go (hunk 8 above)
Comment (n3wscott) [reply pkg/reconciler/clusterducktype/clusterducktype_test.go:32]: yeahyeahyeah :D

File: pkg/reconciler/clusterducktype/testdata/features/smoke.feature (hunk 7 above)
Comment (lionelvillard) [reply pkg/reconciler/clusterducktype/testdata/features/smoke.feature:15]: Actually my question was more about Cucumber itself. I got my answer by looking at it. This is really cool!

**Pull request merged**